lib/src/iterables/enumerate.dart

```diff
@@ -76,10 +76,13 @@ class EnumerateIterator<V> extends Iterator<IndexedValue<V>> {
 
   final Iterator<V> _iterator;
   int _index = 0;
-  IndexedValue<V> _current;
+  IndexedValue<V>/*?*/ _current;
 
   @override
-  IndexedValue<V> get current => _current;
+  IndexedValue<V> get current {
+    final cur = _current;
+    return cur as IndexedValue<V>;
```

Comment:
Why not just `return _current as IndexedValue<V>;`?

Reply:
Hrm I had a vague recollection that I tried that and needed a temp, but maybe I'd made a separate change that conflicted at that time; the above works fine. Done.

Reply:
Ah; it was an unnecessary cast warning.